lib/facter/powershell_version.rb

```diff
@@ -0,0 +1,76 @@
+Facter.add(:powershell_version) do
+  setcode do
+    if Puppet::Util::Platform.windows?
+      version = Facter.value(:powershell_three_version) || Facter.value(:powershell_one_version)
```

Comment:
Facter does not appear to turn it into a `string`, but keep it as a `Gem::Version`. However it does not accept that object as a valid value:

~~~
Warning: Facter: Fact resolution fact='powershell_version', resolution='<anonymous>' resolved to an invalid value: Expected 5.0.10514.6 to be one of [
Integer, Float, TrueClass, FalseClass, NilClass, String, Array, Hash], but was Gem::Version
~~~